spark/datadog_checks/spark/spark.py

```diff
@@ -143,6 +144,8 @@
     'numTotalCompletedBatches': ('spark.streaming.statistics.num_total_completed_batches', MONOTONIC_COUNT),
 }
 
+PROXY_WITH_DIFFERENT_USER_WARNING = re.compile(r'<html>.*<a href="(.*)">.*</html>', re.S)
```

Comment:
I noticed we have `BeautifulSoup` available in this integration already. Would it be worth using it instead of a regex, as in: could it make the parsing more robust?

Reply:
Yeah that's a great idea 👍